### ISSUE TITLE
MarshalIndent very slow and ends up getting killed by OS on large files or when benchmarking

### ISSUE THREAD

**Issue by cwbriscoe** [goccy/go-json#247]
MarshalIndent will crash when benchmarking or writing larger output (~300kb).

go stdlib:
```
go test -bench=BenchmarkSpeed* -benchtime=10s
goos: linux
goarch: amd64
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkSpeed-12    	  223873	     57097 ns/op
PASS
ok  		13.338s
```

goccy/go-json:
```
go test -bench=BenchmarkSpeed* -benchtime=10s
goos: linux
goarch: amd64
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkSpeed-12    	signal: killed
FAIL		8.038s
```

MarshalIndent is very slow compared to the go stdlib.  test.go and input data attached below.

go stdlib:
```
go test
PASS
ok  		0.003s
```

goccy/go-json:
```
go test
PASS
ok  		1.722s
```

Test Code:
```
package main

import (
	//"encoding/json"
	"io/ioutil"
	"testing"
	"time"

	"github.com/goccy/go-json"
)

type Attribute struct {
	Name  string `json:"name"`
	Count int    `json:"count"`
	Key   string `json:"key"`
}

type Attributes struct {
	Facets *Facets `json:"facets,omitempty"`
}

type Campaigns struct {
	ID        string    `json:"id"`
	StartDate time.Time `json:"start_date"`
	EndDate   time.Time `json:"end_date"`
}

type ContentDescriptors struct {
	Description string `json:"description"`
	URL         string `json:"url"`
	Name        string `json:"name"`
}

type ContentInteractiveElement struct {
	Name        string `json:"name"`
	Description string `json:"description"`
}

type ContentRating struct {
	Description  string `json:"description,omitempty"`
	RatingSystem string `json:"rating_system,omitempty"`
	URL          string `json:"url,omitempty"`
}

type Eligibility struct {
	ID          string `json:"id"`
	Operand     string `json:"operand"`
	Operator    string `json:"operator"`
	Name        string `json:"name"`
	Description string `json:"description"`
}

type Entitlement struct {
	Description             string      `json:"description"`
	Duration                int         `json:"duration"`
	ExpAfterFirstUse        int         `json:"exp_after_first_use"`
	FeatureTypeID           int         `json:"feature_type_id"`
	ID                      string      `json:"id"`
	LicenseType             int         `json:"license_type"`
	Metadata                EntMetadata `json:"metadata"`
	Name                    string      `json:"name"`
	PackageType             string      `json:"packageType"`
	Packages                []Package   `json:"packages"`
	PreorderPlaceholderFlag bool        `json:"preorder_placeholder_flag"`
	Size                    uint64      `json:"size"`
	SubType                 int         `json:"subtype"`
	SubtitleLanguageCodes   []string    `json:"subtitle_language_codes"`
	Type                    int         `json:"type"`
	UseCount                int         `json:"use_count"`
	VoiceLanguageCodes      []string    `json:"voice_language_codes"`
}

type EntMetadata struct {
	VoiceLanguageCode    []string `json:"voiceLanguageCode,omitempty"`
	SubtitleLanguageCode []string `json:"subtitleLanguageCode,omitempty"`
}

type Facets struct {
	GameContentType []Attribute `json:"game_content_type,omitempty"`
	SubtitleLang    []Attribute `json:"subtitle_lang,omitempty"`
	Price           []Attribute `json:"price,omitempty"`
	AddonType       []Attribute `json:"addon_type,omitempty"`
	Genre           []Attribute `json:"genre,omitempty"`
	TopCategory     []Attribute `json:"top_category,omitempty"`
	VoiceLang       []Attribute `json:"voice_lang,omitempty"`
	Relationship    []Attribute `json:"relationship,omitempty"`
}

type GameContentType struct {
	Name string `json:"name"`
	Key  string `json:"key"`
}

type Image struct {
	Type int    `json:"type"`
	URL  string `json:"url"`
}

type Package struct {
	PlatformID   int    `json:"platformId"`
	PlatformName string `json:"platformName"`
	Size         uint64 `json:"size"`
}

type Rewards struct {
	ID                 string      `json:"id"`
	Discount           int         `json:"discount"`
	Price              int         `json:"Price"`
	RewardType         int         `json:"reward_type"`
	DisplayPrice       string      `json:"display_price"`
	IsPlus             bool        `json:"isPlus"`
	Campaigns          []Campaigns `json:"campaigns"`
	RewardSourceTypeID int         `json:"reward_source_type_id"`
	StartDate          time.Time   `json:"start_date"`
	EndDate            time.Time   `json:"end_date"`
}

type Sku struct {
	AmortizeFlag                bool          `json:"amortizeFlag"`
	BundleExclusiveFlag         bool          `json:"bundleExclusiveFlag"`
	ChargeImeediatelyFlag       bool          `json:"chargeImmediatelyFlag"`
	ChargeTypeID                int           `json:"charge_type_id"`
	CreditCardRequiredFlag      int           `json:"credit_card_required_flag"`
	DefaultSku                  bool          `json:"defaultSku"`
	DisplayPrice                string        `json:"display_price"`
	Eligibilities               []Eligibility `json:"eligibilities"`
	Entitlements                []Entitlement `json:"entitlements"`
	ID                          string        `json:"id"`
	IsOriginal                  bool          `json:"is_original"`
	Name                        string        `json:"name"`
	Platforms                   []int         `json:"platforms"`
	Price                       int           `json:"price"`
	Rewards                     []Rewards     `json:"rewards"`
	SeasonPassExclusiveFlag     bool          `json:"seasonPassExclusiveFlag"`
	SkuAvailabilityOverrideFlag bool          `json:"skuAvailabilityOverrideFlag"`
	SkuType                     int           `json:"sku_type"`
	Type                        string        `json:"type"`
}

type Game struct {
	AgeLimit                  int                         `json:"age_limit"`
	Attributes                *Attributes                 `json:"attributes,omitempty"`
	Bucket                    string                      `json:"bucket"`
	ContainerType             string                      `json:"container_type"`
	ContentInteractiveElement []ContentInteractiveElement `json:"contentInteractiveElement,omitempty"`
	ContentDesriptors         []ContentDescriptors        `json:"content_descriptors,omitempty"`
	ContentOrigin             int                         `json:"content_origin"`
	ContentRating             ContentRating               `json:"content_rating"`
	ContentType               string                      `json:"content_type"`
	DefaultSku                Sku                         `json:"default_sku"`
	DobRequired               bool                        `json:"dob_required"`
	GameContentTypesList      []GameContentType           `json:"gameContentTypesList"`
	GameContentType           string                      `json:"game_contentType"`
	ID                        string                      `json:"id"`
	Images                    []Image                     `json:"images"`
	LegalText                 string                      `json:"legal_text,omitempty"`
	Links                     []Game                      `json:"links"`
	LongDesc                  string                      `json:"long_desc"`
	Name                      string                      `json:"name"`
	PageTypeID                int                         `json:"pageTypeId"`
	ParentLinks               []Game                      `json:"parent_links"`
	ParentName                string                      `json:"parent_name"`
	PlayablePlatform          []string                    `json:"playable_platform"`
	ProviderName              string                      `json:"provider_name"`
	ReleaseDate               time.Time                   `json:"release_date"`
	Restricted                bool                        `json:"restricted"`
	Revision                  int                         `json:"revision"`
	ShortName                 string                      `json:"short_name"`
	Size                      uint64                      `json:"size"`
	Skus                      []Sku                       `json:"skus"`
	Start                     int                         `json:"start"`
	Timstamp                  uint64                      `json:"timestamp"`
	TitleName                 string                      `json:"title_name"`
	TopCategory               string                      `json:"top_category"`
	TotalResults              int                         `json:"total_results"`
	URL                       string                      `json:"url,omitempty"`
	Etag                      string                      `json:"etag"`
}

func TestMarshalIndent(t *testing.T) {
	pkg := &Game{}

	data, err := ioutil.ReadFile("./testin.json")
	if err != nil {
		t.Fatalf("ReadFile Failed: %v", err)
	}

	err = json.Unmarshal(data, &pkg)
	if err != nil {
		t.Fatalf("Unmarshal Failed: %v", err)
	}

	data, err = json.MarshalIndent(pkg, "", "  ")
	if err != nil {
		t.Fatalf("MarshalIndent Failed: %v", err)
	}

	err = ioutil.WriteFile("./file.json", data, 0644)
	if err != nil {
		t.Fatalf("WriteFile Failed: %v", err)
	}

}

func BenchmarkSpeed(b *testing.B) {

	b.RunParallel(func(pb *testing.PB) {
		data, _ := ioutil.ReadFile("./testin.json")
		pkg := &Game{}
		json.Unmarshal(data, &pkg)
		for pb.Next() {
			json.MarshalIndent(pkg, "", "  ")
		}
	})
}
```

Test File:
```
{"age_limit":13,"attributes":{"facets":{"price":[{"name":"Free","count":1,"key":"0-0"},{"name":"Under $1.99","count":1,"key":"0-199"}],"top_category":[{"name":"Add-On","count":1,"key":"add_on"}],"relationship":[{"name":"Add-Ons","count":1,"key":"addons"}]},"next":[]},"bucket":"games","container_type":"product","contentInteractiveElement":[{"name":"In-Game Purchases","description":"In-Game Purchases"},{"name":"Users Interact","description":"Users Interact"}],"content_descriptors":[{"description":"Violence","url":null,"name":"Violence"},{"description":"Blood","url":null,"name":"Blood"}],"content_origin":0,"content_rating":{"description":"ESRB Teen","rating_system":"ESRB","url":"https://cdn-a.sonyentertainmentnetwork.com/grc/images/ratings/hd/esrb/t.png"},"content_type":"1","default_sku":{"amortizeFlag":false,"bundleExclusiveFlag":false,"chargeImmediatelyFlag":false,"charge_type_id":0,"credit_card_required_flag":0,"defaultSku":true,"display_price":"Free","eligibilities":[],"entitlements":[{"description":null,"drms":[],"duration":0,"durationOverrideTypeId":null,"exp_after_first_use":0,"feature_type_id":1,"id":"UP4844-CUSA14106_00-CARAVANSTORIES00","license_type":0,"metadata":null,"name":"CARAVAN STORIES","packageType":"PS4GD","packages":[{"platformId":13,"platformName":"ps4","size":8676179968}],"preorder_placeholder_flag":false,"size":0,"subType":0,"subtitle_language_codes":null,"type":5,"use_count":0,"voice_language_codes":null}],"id":"UP4844-CUSA14106_00-CARAVANSTORIES00-U099","is_original":false,"name":"Downloadable Game","platforms":[1,18,10,13],"price":0,"rewards":[],"seasonPassExclusiveFlag":false,"skuAvailabilityOverrideFlag":false,"sku_type":0,"type":"standard"},"dob_required":false,"gameContentTypesList":[{"name":"Full Game","key":"FULL_GAME"}],"game_contentType":"Full Game","id":"UP4844-CUSA14106_00-CARAVANSTORIES00","images":[{"type":10,"url":"https://vulcan.dl.playstation.net/img/cfn/11307HjHfX930QKajl-BxnwCvnmmnkPSx9cU_ioB8hOKOBDE-swj6UZa6LabYvm2BXWCsk5JjV2MtPvyi9bX-Ck0RPkze8qq.png"},{"type":12,"url":"https://vulcan.dl.playstation.net/img/cfn/11307kXs8SEYO8yUUVcXo4AD_EPjRuS0s1G-39Qng889Ts5gYDczYapunXvp6rhbKUEI4vhqCist3v1oAm32EwwagukJY_Qv.jpg"}],"legal_text":"<br/>© Aiming Inc. <br/>","links":[{"bucket":"games","container_type":"product","content_type":"1","default_sku":{"amortizeFlag":false,"bundleExclusiveFlag":false,"chargeImmediatelyFlag":false,"charge_type_id":0,"credit_card_required_flag":0,"defaultSku":true,"display_price":"Free","eligibilities":[{"id":"UP4844-CUSA14106_00-PP2018","operand":"EXISTS","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]},{"id":"UP4844-CUSA14106_00-PP2018","operand":"IS_ACTIVE","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]}],"end_date":"2021-07-01T04:00:00Z","entitlements":[{"description":null,"drms":[],"duration":0,"durationOverrideTypeId":null,"exp_after_first_use":0,"feature_type_id":0,"id":"UP4844-CUSA14106_00-PP2018","license_type":0,"metadata":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","packageType":null,"packages":[],"preorder_placeholder_flag":false,"size":0,"subType":0,"subtitle_language_codes":null,"type":1,"use_count":1,"voice_language_codes":null}],"id":"UP4844-CUSA14106_00-PP20180000000000-U001","is_original":false,"name":"Virtual Currency","platforms":[0,18,10,13],"price":0,"rewards":[{"id":"IP9101-NPIA90005_01-RWD-975","entitlement_id":"IP9101-NPIA90005_01-RWD-975","service_provider_id":"IP9101","discount":0,"price":0,"reward_type":3,"display_price":"Free","name":"SP Exclusive/Forever; 0% off","description":"<b><font color=\"#FDD017\">This is an exclusive PlayStation®Plus offer.</font></b>","isPlus":true,"rewardSourceId":1,"reward_source_type_id":1,"start_date":"2021-06-01T04:00:00Z","end_date":"2021-07-01T04:00:00Z"}],"seasonPassExclusiveFlag":false,"skuAvailabilityOverrideFlag":false,"sku_type":0,"type":"standard"},"gameContentTypesList":[{"name":"Add-On","key":"ADD-ON"}],"game_contentType":"Add-On","id":"UP4844-CUSA14106_00-PP20180000000000","images":[],"name":"CARAVAN STORIES-【Bonus for PS Plus】Phantasm GEM 500(June2021)","parent_name":"CARAVAN STORIES","playable_platform":["PS4™"],"provider_name":"AIMING INC.","release_date":"2021-06-01T00:00:00Z","restricted":false,"revision":0,"short_name":"【Bonus for PS Plus】Phantasm GEM 500(June2021)","timestamp":1623177463000,"top_category":"add_on","url":"https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP4844-CUSA14106_00-PP20180000000000/1623177463"},{"age_limit":13,"attributes":{"facets":{"price":[{"name":"Free","count":1,"key":"0-0"},{"name":"Under $1.99","count":1,"key":"0-199"}],"top_category":[{"name":"Add-On","count":1,"key":"add_on"}],"relationship":[{"name":"Add-Ons","count":1,"key":"addons"}]},"next":[]},"bucket":"games","container_type":"container","contentInteractiveElement":[{"name":"In-Game Purchases","description":"In-Game Purchases"},{"name":"Users Interact","description":"Users Interact"}],"content_descriptors":[{"description":"Violence","url":null,"name":"Violence"},{"description":"Blood","url":null,"name":"Blood"}],"content_origin":0,"content_rating":{"description":"ESRB Teen","rating_system":"ESRB","url":"https://cdn-a.sonyentertainmentnetwork.com/grc/images/ratings/hd/esrb/t.png"},"content_type":"1","dob_required":false,"gameContentTypesList":[{"name":"Full Game","key":"FULL_GAME"}],"game_contentType":"Full Game","id":"UP4844-CUSA14106_00-CARAVANSTORIES00:ADD-ONS","legal_text":"<br/>© Aiming Inc. <br/>","links":[{"bucket":"games","container_type":"product","content_type":"1","default_sku":{"amortizeFlag":false,"bundleExclusiveFlag":false,"chargeImmediatelyFlag":false,"charge_type_id":0,"credit_card_required_flag":0,"defaultSku":true,"display_price":"Free","eligibilities":[{"id":"UP4844-CUSA14106_00-PP2018","operand":"EXISTS","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]},{"id":"UP4844-CUSA14106_00-PP2018","operand":"IS_ACTIVE","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]}],"end_date":"2021-07-01T04:00:00Z","entitlements":[{"description":null,"drms":[],"duration":0,"durationOverrideTypeId":null,"exp_after_first_use":0,"feature_type_id":0,"id":"UP4844-CUSA14106_00-PP2018","license_type":0,"metadata":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","packageType":null,"packages":[],"preorder_placeholder_flag":false,"size":0,"subType":0,"subtitle_language_codes":null,"type":1,"use_count":1,"voice_language_codes":null}],"id":"UP4844-CUSA14106_00-PP20180000000000-U001","is_original":false,"name":"Virtual Currency","platforms":[0,18,10,13],"price":0,"rewards":[{"id":"IP9101-NPIA90005_01-RWD-975","entitlement_id":"IP9101-NPIA90005_01-RWD-975","service_provider_id":"IP9101","discount":0,"price":0,"reward_type":3,"display_price":"Free","name":"SP Exclusive/Forever; 0% off","description":"<b><font color=\"#FDD017\">This is an exclusive PlayStation®Plus offer.</font></b>","isPlus":true,"rewardSourceId":1,"reward_source_type_id":1,"start_date":"2021-06-01T04:00:00Z","end_date":"2021-07-01T04:00:00Z"}],"seasonPassExclusiveFlag":false,"skuAvailabilityOverrideFlag":false,"sku_type":0,"type":"standard"},"gameContentTypesList":[{"name":"Add-On","key":"ADD-ON"}],"game_contentType":"Add-On","id":"UP4844-CUSA14106_00-PP20180000000000","images":[],"name":"CARAVAN STORIES-【Bonus for PS Plus】Phantasm GEM 500(June2021)","parent_name":"CARAVAN STORIES","playable_platform":["PS4™"],"provider_name":"AIMING INC.","release_date":"2021-06-01T00:00:00Z","restricted":false,"revision":0,"short_name":"【Bonus for PS Plus】Phantasm GEM 500(June2021)","timestamp":1623177463000,"top_category":"add_on","url":"https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP4844-CUSA14106_00-PP20180000000000/1623177463","skus":[{"amortizeFlag":false,"bundleExclusiveFlag":false,"chargeImmediatelyFlag":false,"charge_type_id":0,"credit_card_required_flag":0,"defaultSku":true,"display_price":"Free","eligibilities":[{"id":"UP4844-CUSA14106_00-PP2018","operand":"EXISTS","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]},{"id":"UP4844-CUSA14106_00-PP2018","operand":"IS_ACTIVE","operator":"FALSE","rightOperand":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","description":null,"entitlement_type":null,"drms":[]}],"end_date":"2021-07-01T04:00:00Z","entitlements":[{"description":null,"drms":[],"duration":0,"durationOverrideTypeId":null,"exp_after_first_use":0,"feature_type_id":0,"id":"UP4844-CUSA14106_00-PP2018","license_type":0,"metadata":null,"name":"【Bonus for PS Plus】Phantasm Gem 500 (June2021)","packageType":null,"packages":[],"preorder_placeholder_flag":false,"size":0,"subType":0,"subtitle_language_codes":null,"type":1,"use_count":1,"voice_language_codes":null}],"id":"UP4844-CUSA14106_00-PP20180000000000-U001","is_original":false,"name":"Virtual Currency","platforms":[0,18,10,13],"price":0,"rewards":[{"id":"IP9101-NPIA90005_01-RWD-975","entitlement_id":"IP9101-NPIA90005_01-RWD-975","service_provider_id":"IP9101","discount":0,"price":0,"reward_type":3,"display_price":"Free","name":"SP Exclusive/Forever; 0% off","description":"<b><font color=\"#FDD017\">This is an exclusive PlayStation®Plus offer.</font></b>","isPlus":true,"rewardSourceId":1,"reward_source_type_id":1,"start_date":"2021-06-01T04:00:00Z","end_date":"2021-07-01T04:00:00Z"}],"seasonPassExclusiveFlag":false,"skuAvailabilityOverrideFlag":false,"sku_type":0,"type":"standard"}],"skuFound":true}],"mediaList":{"previews":[{"type":"PREVIEW","typeId":2,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307sq-3VGni4buq6CKAE3IEIJimASaXv0xrxcKd1V9Mz-wzUoap-2CFZQk0a5V8N-91bedZhK7oYCFnJbULnzN96YKLtR6.mp4","order":0}],"screenshots":[{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307oomH1WZ0QeQd6kQ7_OaE8AGga9WOinKl_XZPhWwZEzM_rim9YZFT6DNp19WSzICYpdLmFj6o2kttIpyHLPRZxouFgI-.jpg","order":0},{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307dkTR1Xv6WO-Qf0m7UCYLm4hIY9hriyVKaZE32lHHwIoT2M-mONRMVu2KBysh6-Sq3SP5pX67ekGjxXloRDBETozgArI.jpg","order":1},{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307AOroFwrTO-EKdVrwJvswLRo4fNjdIzvGu8LjM-F5MtA-EssBidwyFtu_BuvTJ8wpSFuP97mJON3ln1i-ZZtkZwoILtw.jpg","order":2}]},"metadata":{"cn_remotePlay":{"name":"cn_remotePlay","values":["TRUE"]},"cn_vrEnabled":{"name":"cn_vrEnabled","values":["FALSE"]},"cn_playstationMove":{"name":"cn_playstationMove","values":["NOTREQUIRED"]},"cn_numberOfNetworkPlayers":{"name":"cn_numberOfNetworkPlayers","values":["21"]},"secondary_classification":{"name":"secondary_classification","values":["GAME"]},"cn_vrRequired":{"name":"cn_vrRequired","values":["FALSE"]},"game_genre":{"name":"Genre","values":["ROLE_PLAYING_GAMES"]},"cn_psEnhanced":{"name":"cn_psEnhanced","values":["FALSE"]},"playable_platform":{"name":"playable_platform","values":["PS4™"]},"cn_numberOfPlayers":{"name":"cn_numberOfPlayers","values":["1"]},"cn_dualshockVibration":{"name":"cn_dualshockVibration","values":["TRUE"]},"tertiary_classification":{"name":"tertiary_classification","values":["FREEMIUM"]},"container_type":{"name":"Container Type","values":["NP_PS4_REPRESENTATIVE"]},"genre":{"name":"Genre","values":["Role-Playing Games (RPG)"]},"cn_inGamePurchases":{"name":"cn_inGamePurchases","values":["NOTREQUIRED"]},"cn_psVrAimRequired":{"name":"cn_psVrAimRequired","values":["FALSE"]},"cn_playstationCamera":{"name":"cn_playstationCamera","values":["NOTREQUIRED"]},"cn_singstarMicrophone":{"name":"cn_singstarMicrophone","values":["NOTREQUIRED"]},"cn_crossPlatformPSVita":{"name":"cn_crossPlatformPSVita","values":["FALSE"]},"cn_onlinePlayMode":{"name":"cn_onlinePlayMode","values":["REQUIRED"]},"cn_psVrAimEnabled":{"name":"cn_psVrAimEnabled","values":["FALSE"]},"primary_classification":{"name":"primary_classification","values":["PREMIUM_GAME"]}},"name":"CARAVAN STORIES","pageTypeId":2,"provider_name":"AIMING INC.","release_date":"2019-09-10T00:00:00Z","restricted":false,"revision":22,"short_name":"CARAVAN STORIES","size":1,"sort":"static","star_rating":{"total":"2819","score":"3.78","count":[{"star":1,"count":577},{"star":2,"count":109},{"star":3,"count":240},{"star":4,"count":335},{"star":5,"count":1558}]},"start":0,"timestamp":1622141321000,"title_name":"CARAVAN STORIES","top_category":"downloadable_game","total_results":1,"url":"https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP4844-CUSA14106_00-CARAVANSTORIES00?relationship=ADD-ONS","link_type":"relationship","container_relationship":"ADD-ONS","key_name":"add-ons","expanded":true}],"long_desc":"<b>To play this game on PS5, your system may need to be updated to the latest system software. Although this game is playable on PS5, some features available on PS4 may be absent. See PlayStation.com/bc for more details.</b><br>Live for the journey!<br/><br/><br/>Do you ever find yourself with a similar feeling of wanderlust?<br/>Vast open prairies, mystical forests as far as the eye can see, barren lands covered in rocky outcroppings...all places to visit alongside your friends and allies.<br/>Caravan Stories is where you can find exactly what you're looking for.<br/><br/><br/>First, pick what race you'd like to go on your adventure as.<br/>Entice characters into joining you on your adventure, then explore the endless landscapes to your heart's content.<br/><br/><br/>Time spent adventuring in your lively Caravan surrounded by your many allies is surely an exciting experience you won't soon forget.<br/><br/><br/>Come, your adventure awaits!!<br/><br/><br/>3D Display with compatible 3D glasses (sold separately) required for 3D features.   <br/><br/><br/><br/><br/><br/><br/>Offline single player<br/>Online multiplayer (2-21 players). Online features require an account and are subject to our terms of service (playstationnetwork.com/terms-of-service), our privacy policy (playstationnetwork.com/privacy-policy), and the game publisher’s privacy policy.<br/>Uses DUALSHOCK 4 vibration function<br/>Remote Play allows users to connect to and play games on their PlayStation®4 or PlayStation®5 using a mobile device, PC, or another PlayStation®4 or PlayStation®5.<br/>Online play required<br/>Stereoscopic 3D Game<br/><br><br>Online features require an account and are subject to terms of service and applicable privacy policy (playstationnetwork.com/terms-of-service & playstationnetwork.com/privacy-policy). <br><br>Software subject to license (us.playstation.com/softwarelicense). <br><br>One-time license fee for play on account’s designated primary PS4™ system and other PS4™ systems when signed in with that account.<br>","mediaList":{"previews":[{"type":"PREVIEW","typeId":2,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307sq-3VGni4buq6CKAE3IEIJimASaXv0xrxcKd1V9Mz-wzUoap-2CFZQk0a5V8N-91bedZhK7oYCFnJbULnzN96YKLtR6.mp4","order":0}],"screenshots":[{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307oomH1WZ0QeQd6kQ7_OaE8AGga9WOinKl_XZPhWwZEzM_rim9YZFT6DNp19WSzICYpdLmFj6o2kttIpyHLPRZxouFgI-.jpg","order":0},{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307dkTR1Xv6WO-Qf0m7UCYLm4hIY9hriyVKaZE32lHHwIoT2M-mONRMVu2KBysh6-Sq3SP5pX67ekGjxXloRDBETozgArI.jpg","order":1},{"type":"SCREENSHOT","typeId":4,"source":"PRODUCT_MEDIA","url":"https://vulcan.dl.playstation.net/img/cfn/11307AOroFwrTO-EKdVrwJvswLRo4fNjdIzvGu8LjM-F5MtA-EssBidwyFtu_BuvTJ8wpSFuP97mJON3ln1i-ZZtkZwoILtw.jpg","order":2}]},"media_layouts":[{"type":"description","height":1,"width":4}],"metadata":{"cn_remotePlay":{"name":"cn_remotePlay","values":["TRUE"]},"cn_vrEnabled":{"name":"cn_vrEnabled","values":["FALSE"]},"cn_playstationMove":{"name":"cn_playstationMove","values":["NOTREQUIRED"]},"cn_numberOfNetworkPlayers":{"name":"cn_numberOfNetworkPlayers","values":["21"]},"secondary_classification":{"name":"secondary_classification","values":["GAME"]},"cn_vrRequired":{"name":"cn_vrRequired","values":["FALSE"]},"game_genre":{"name":"Genre","values":["ROLE_PLAYING_GAMES"]},"cn_psEnhanced":{"name":"cn_psEnhanced","values":["FALSE"]},"playable_platform":{"name":"playable_platform","values":["PS4™"]},"cn_numberOfPlayers":{"name":"cn_numberOfPlayers","values":["1"]},"cn_dualshockVibration":{"name":"cn_dualshockVibration","values":["TRUE"]},"tertiary_classification":{"name":"tertiary_classification","values":["FREEMIUM"]},"container_type":{"name":"Container Type","values":["NP_PS4_REPRESENTATIVE"]},"genre":{"name":"Genre","values":["Role-Playing Games (RPG)"]},"cn_inGamePurchases":{"name":"cn_inGamePurchases","values":["NOTREQUIRED"]},"cn_psVrAimRequired":{"name":"cn_psVrAimRequired","values":["FALSE"]},"cn_playstationCamera":{"name":"cn_playstationCamera","values":["NOTREQUIRED"]},"cn_singstarMicrophone":{"name":"cn_singstarMicrophone","values":["NOTREQUIRED"]},"cn_crossPlatformPSVita":{"name":"cn_crossPlatformPSVita","values":["FALSE"]},"cn_onlinePlayMode":{"name":"cn_onlinePlayMode","values":["REQUIRED"]},"cn_psVrAimEnabled":{"name":"cn_psVrAimEnabled","values":["FALSE"]},"primary_classification":{"name":"primary_classification","values":["PREMIUM_GAME"]}},"name":"CARAVAN STORIES","pageTypeId":2,"playable_platform":["PS4™"],"promomedia":[{"id":0,"key":"preview","type":2,"url":"https://vulcan.dl.playstation.net/img/cfn/11307sq-3VGni4buq6CKAE3IEIJimASaXv0xrxcKd1V9Mz-wzUoap-2CFZQk0a5V8N-91bedZhK7oYCFnJbULnzN96YKLtR6.mp4?country=US"}],"provider_name":"AIMING INC.","release_date":"2019-09-10T00:00:00Z","restricted":false,"revision":22,"short_name":"CARAVAN STORIES","size":2,"sku_links":[],"skus":[{"amortizeFlag":false,"bundleExclusiveFlag":false,"chargeImmediatelyFlag":false,"charge_type_id":0,"credit_card_required_flag":0,"defaultSku":true,"display_price":"Free","eligibilities":[],"entitlements":[{"description":null,"drms":[],"duration":0,"durationOverrideTypeId":null,"exp_after_first_use":0,"feature_type_id":1,"id":"UP4844-CUSA14106_00-CARAVANSTORIES00","license_type":0,"metadata":null,"name":"CARAVAN STORIES","packageType":"PS4GD","packages":[{"platformId":13,"platformName":"ps4","size":8676179968}],"preorder_placeholder_flag":false,"size":0,"subType":0,"subtitle_language_codes":null,"type":5,"use_count":0,"voice_language_codes":null}],"id":"UP4844-CUSA14106_00-CARAVANSTORIES00-U099","is_original":false,"name":"Downloadable Game","platforms":[0,18,10,13],"price":0,"rewards":[],"seasonPassExclusiveFlag":false,"skuAvailabilityOverrideFlag":false,"sku_type":0,"type":"standard"}],"sort":"static","star_rating":{"total":"2819","score":"3.78","count":[{"star":1,"count":577},{"star":2,"count":109},{"star":3,"count":240},{"star":4,"count":335},{"star":5,"count":1558}]},"start":0,"timestamp":1622141321000,"title_name":"CARAVAN STORIES","top_category":"downloadable_game","total_results":2,"url":"https://store.playstation.com/store/api/chihiro/00_09_000/container/US/en/19/UP4844-CUSA14106_00-CARAVANSTORIES00"}
```

**Comment by goccy** [comment]
There is no API called UnmarshalIndent in my library. Also, write the benchmark code directly here, not in a zip file.

**Comment by cwbriscoe** [comment]
Sorry, I meant MarshalIndent.  I pasted in the test code and the input file.

**Comment by goccy** [comment]
OK, I fixed this issue with #249
